csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -990,79 +990,83 @@ void HopperMultipleMatmulScheduler::scheduleOutputTensor(TensorView* c) {
+        // -> [..., Mo, No, Mio, Nio, Mii, Nii]
+        tv->reorder({{-3, -2}});
+        tv->merge(-4);
+        auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
+            tv->getLoopDomain());
+        tv->setLoopDomain(s.as<IterDomain*>());
+        tv->axis(-5)->parallelize(ParallelType::TIDy);
+      }
+      parallelizeBlocks({dc});
+      parallelizeBlocks({d});
       d->axis(-1)->parallelize(ParallelType::Vectorize);
-
-      // Propagate output tensor transformations back to smem_epilogue
-      scheduler_utils::BoundedDirectionalTransformPropagator::backward(
-          d, -1, smem_epilogues_);
     }
   } else {
-    for (TensorView* mma_result : mma_results_) {
-      scheduler_utils::BoundedDirectionalTransformPropagator::forward(
-          mma_result,
-          -1,
-          output_tvs,
-          scheduler_utils::BoundedDirectionalTransformPropagator::Options()
-              .propagateParallelType()
-              .propagateToBoundary());
-    }
-    for (auto [dc, d] : cached_outputs_) {
-      // We might propagate an inner dimension that is not compatible with the
-      // output or bias-like inputs. In those cases, we will further split
-      // this dimension with an outer unrolled loop to achieve the proper
-      // vectorization as specified by params.supported_vec_size.epilogue.
-      NVF_ERROR(d->axis(-1)->extent()->isConst());
-      int64_t d_extent = d->axis(-1)->extent()->value().as<int64_t>();
-      if (d_extent > params_->supported_vec_size.epilogue) {
-        // Should always be a divisible split
-        NVF_ERROR(d_extent % params_->supported_vec_size.epilogue == 0);
-        d->split(
-            -1, params_->supported_vec_size.epilogue, /*inner_split=*/true);
-        d->axis(-2)->parallelize(ParallelType::Unroll);
+    // TODO: Use stmatrix to load from registers to shared memory
+    constexpr int64_t stmatrix_tile_m = 16;
+    constexpr int64_t stmatrix_tile_n = 16;
+    constexpr int64_t tma_m = 16;
+    constexpr int64_t tma_n = 64;
+
+    // Manually schedule register cache and output TensorView
+    for (auto& [dc, d] : cached_outputs_) {
+      // NOTE: cacheBefore does not work with blockTileTensors
+      // Using cacheAfter for now
+      TensorView* d_smem = cacheAfter(dc, LoadStoreOpType::Set);
+
+      // Set MemoryType
+      dc->setMemoryType(MemoryType::Local);
+      d_smem->setMemoryType(MemoryType::Shared);
+
+      // Set LoadStoreOp
+      // TODO Use LoadStoreOpType::StMatrix on d_smem definition
+      d->definition()->as<LoadStoreOp>()->setOpType(
+          LoadStoreOpType::CpAsyncBulkTensorTile);
+
+      // Block Schedule
+      blockTileTensors({dc});
+      blockTileTensors({d_smem});
+      blockTileTensors({d});
+
+      parallelizeBlocks({dc});
+      parallelizeBlocks({d_smem});
+      parallelizeBlocks({d});
+
+      // Apply common transformation
+      for (auto tv : {dc, d_smem, d}) {
+        // Original [..., Mo, No]
```

Comment:
Maybe start with this
```suggestion
        // Original [..., Mo, No, Mi, Ni]
```
Then it will make more sense to have `Mii` and `Mio` etc. below